pkg/ssh/ssh.go

```diff
@@ -377,15 +377,18 @@ func (l *SSHTunnelList) healthCheck(e sshTunnelEntry) error {
 func (l *SSHTunnelList) removeAndReAdd(e sshTunnelEntry) {
 	// Find the entry to replace.
 	l.tunnelsLock.Lock()
-	defer l.tunnelsLock.Unlock()
 	for i, entry := range l.entries {
 		if entry.Tunnel == e.Tunnel {
 			l.entries = append(l.entries[:i], l.entries[i+1:]...)
 			l.adding[e.Address] = true
-			go l.createAndAddTunnel(e.Address)
-			return
+			break
 		}
 	}
+	l.tunnelsLock.Unlock()
+	if err := e.Tunnel.Close(); err != nil {
```

Comment:
It is safe to call 'Close()` (but it was worthwhile for me to revisit the logic here).

`e` is the entry we have already decided we want to pull out of the list (if present) and close. It is an `sshTunnelEntry` (not a pointer) so dereferencing `e` is okay, and `e.Tunnel` won't be nil. If `e.Tunnel` was never set/opened, `Close()` will return an error.